src/NitroSharp/Content/ArchiveManager.cs

```diff
@@ -0,0 +1,255 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.IO.MemoryMappedFiles;
+using System.Text;
+
+namespace NitroSharp.Content
+{
+    internal interface IArchiveFile : IDisposable
+    {
+        public void OpenArchive();
+        public Stream OpenStream(string path);
+        public bool Contains(string path);
+    }
+
+    internal class VFSNode : IDisposable
+    {
+        public List<IArchiveFile>? MountedArchives { get; set; }
+        public Dictionary<string, VFSNode>? Children { get; set; }
```

Comment:
Unfortunately, I'm not yet familiar with NPA and AFS, so I can't really tell if it needs to be this complicated.

Reply:
This has nothing to do with the archive formats themselves, it's the cleanest way I had found to accomodate the weird edge cases (like 2 archives mounted at the same place). This structure is quite expensive to build because it uses a lot of allocation but it's only done once and should be pretty fast to use during the game. If you've got any better idea for this, I will defenitly implement it, I'm bad at data structures.